DiscordBot/Services/BirthdayAnnouncementService.cs

```diff
@@ -0,0 +1,266 @@
+                {
+                    // Check if this birthday is today (ignoring year)
+                    if (birthDate.Month == today.Month && birthDate.Day == today.Day)
+                    {
+                        var age = CalculateAge(birthDate, today);
+                        birthdays.Add(new BirthdayInfo { Name = name, BirthDate = birthDate, Age = age });
+                    }
+                }
+            }
+        }
+        catch (Exception e)
+        {
+            _loggingService.LogAction($"[{ServiceName}] Error fetching birthday data: {e.Message}", ExtendedLogSeverity.LowWarning);
+        }
+        
+        return birthdays;
+    }
+    
+    private bool TryParseBirthdayDate(string dateString, string yearString, out DateTime birthDate)
+    {
+        birthDate = default;
+        
+        try
+        {
+            var provider = CultureInfo.InvariantCulture;
+            
+            // Add year if available and not empty
+            if (!string.IsNullOrEmpty(yearString) && !yearString.Contains("&nbsp;"))
+            {
+                dateString = $"{dateString}/{yearString.Trim()}";
+                birthDate = DateTime.ParseExact(dateString, "M/d/yyyy", provider);
+            }
+            else
+            {
+                // Parse without year, assume current year for calculation
+                var tempDate = DateTime.ParseExact(dateString, "M/d", provider);
+                birthDate = new DateTime(DateTime.Today.Year, tempDate.Month, tempDate.Day);
+            }
+            
+            return true;
+        }
+        catch (FormatException)
+        {
+            return false;
+        }
+    }
+    
+    private int? CalculateAge(DateTime birthDate, DateTime today)
+    {
+        if (birthDate.Year == today.Year)
+        {
+            return null; // No year information available
+        }
+        
+        var age = today.Year - birthDate.Year;
+        if (today.Month < birthDate.Month || (today.Month == birthDate.Month && today.Day < birthDate.Day))
+        {
+            age--;
+        }
+        
+        return age;
+    }
+    
+    private string FormatBirthdayAnnouncement(BirthdayInfo birthday)
+    {
+        var message = $"🎉 **Happy Birthday {birthday.Name}!** 🎂";
+        
+        if (birthday.Age.HasValue)
+        {
+            message += $" Hope you have a wonderful {GetAgeOrdinal(birthday.Age.Value)} birthday!";
+        }
+        else
+        {
+            message += " Hope you have a wonderful day!";
+        }
+        
+        return message;
+    }
+    
+    private string GetAgeOrdinal(int age)
+    {
+        // Handle special cases for 11th, 12th, 13th regardless of tens digit
+        var lastTwoDigits = age % 100;
+        if (lastTwoDigits >= 11 && lastTwoDigits <= 13)
+        {
+            return $"{age}th";
+        }
+        
+        return age % 10 switch
```

Comment:
The build pipeline reports an error for this line:
`#10 3.090 /app/Services/BirthdayAnnouncementService.cs(243,16): error CS0019: Operator '%' cannot be applied to operands of type 'int' and 'string'` #